### PR TITLE
Beams with variable second moment(or elastic modulus) can now be solved

### PR DESCRIPTION
**Pull request by jashan498**
#### Brief description of what is fixed or changed
Example:
variable_momemt
```
>>> E, I = symbols('E, I', positive=True)
>>> var('x')
x
>>> b = Beam(4, E, I*(8 - x)/4, x)
>>> b.apply_load(20, 4, -1)
>>> var('R M')
(R, M)
>>> b.apply_load(R, 0, -1)
>>> b.apply_load(M, 0, -2)
>>> b.bc_deflection = [(0, 0)]
>>> b.bc_slope = [(0, 0)]
>>> b.solve_for_reaction_loads(R, M)
>>> b.slope().subs(x, 4).evalf()
98.1929022208175/(E*I)
>>> b.deflection().subs(x, 4).evalf()
247.22839111673/(E*I)
```

#### Other comments
@parsoyaarihant @moorepants please have a look at the test case I edited(I think it was dimensionally wrong before).
#### Release Notes

* physics.continuum_mechanics
  * Beams with variable second moment(or elastic modulus) can now be solved

### CODE REVIEW

File: sympy/physics/continuum_mechanics/tests/test_beam.py [hunk 1]
```diff
@@ -222,6 +222,24 @@ def test_beam_units():
...
+    b.apply_load(20, 4, -1)
+    R, M = symbols('R, M')
+    b.apply_load(R, 0, -1)
+    b.apply_load(M, 0, -2)
+    b.bc_deflection = [(0, 0)]
+    b.bc_slope = [(0, 0)]
+    b.solve_for_reaction_loads(R, M)
+    assert b.slope().subs(x, 4)*E*I == 4*(-80*log(8) + 80 + 80*log(4))
+    assert b.deflection().subs(x, 4)*E*I == 4*(-960*log(2) - 320*log(4)
+        - 160 - 320*meijerg(((3, 1), ()), ((), (2, 0)), 1) + 640*log(8))
```
Comment (moorepants) [inline sympy/physics/continuum_mechanics/tests/test_beam.py]: I'd be nice if you just compared directly to the final expression. Why is meijerg used?

**Comment by moorepants** [comment]
Docstring updates to show this example (or another one) and clarification on limitations are needed. I'd like to see E(x) working too in a test.

File: sympy/physics/continuum_mechanics/tests/test_beam.py [hunk 2]
```diff
@@ -131,13 +131,13 @@ def test_Beam():
 
     # Test for slope distribution function
     p = b2.slope()
-    q = (f - w0*SingularityFunction(e, a1, 4)/24 + w0*SingularityFunction(x, a1, 4)/24 - w2*SingularityFunction(e, c1, 2)/2 + w2*SingularityFunction(x, c1, 2)/2)
-    assert p == q/(E*I)
+    q = (w0*SingularityFunction(x, a1, 4)/24 + w2*SingularityFunction(x, c1, 2)/2)/(E*I) + (E*I*f - w0*SingularityFunction(e, a1, 4)/24 - w2*SingularityFunction(e, c1, 2)/2)/(E*I)
+    assert p == q
```
[the remark this answers is not in the document]
Comment (jashan498) [reply sympy/physics/continuum_mechanics/tests/test_beam.py]: [This](https://github.com/sympy/sympy/blob/56d706fd0dbbbc6abae4380ece8e208c16475dc1/sympy/physics/continuum_mechanics/beam.py#L982) equation is used to calculate constants in current master. But I think the correct relation is:



So, thatswhy I think `slope_curve = integrate(S(1)/(E*I)*self.bending_moment(), x) + C3` should be used there instead.

[the remark this answers is not in the document]
Comment (jashan498) [reply sympy/physics/continuum_mechanics/tests/test_beam.py]: In most cases, `bc_slope` is like `[(x, 0)]`, due to which `slope_curve` becomes zero (making `C3` equal to zero). That is the reason this change only affected 2 test cases.

File: sympy/physics/continuum_mechanics/tests/test_beam.py (hunk 1 above)
Comment (jashan498) [reply sympy/physics/continuum_mechanics/tests/test_beam.py]: `meijerg` appeared due to `integrate` function. This is the same example shown in above description.

**Comment by jashan498** [comment]
ping @moorepants

**Comment by jashan498** [comment]
@moorepants what else should I change in this PR?

File: sympy/physics/continuum_mechanics/tests/test_beam.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (moorepants) [reply sympy/physics/continuum_mechanics/tests/test_beam.py]: ok

**Comment by moorepants** [comment]
Great work!

**Pull request merged**